index.html

```diff
@@ -1012,6 +1019,160 @@ <h3>
+          system.
+        </p>
+        <p>
+          To <dfn>process the `file_handlers` member</dfn>, given [=ordered
+          map=] |json:ordered map|, [=ordered map=] |manifest:ordered map|,
+          |manifest_url::URL|:
+        </p>
+        <ol class="algorithm">
+          <li>Let |processedFileHandlers:list| be a new [=list=].
+          </li>
+          <li>Set |manifest|["file_handlers"] to |processedFileHandlers|.
+          </li>
+          <li>If |json|["file_handlers"] doesn't [=map/exist=] or
+          |json|["file_handlers"] is not a [=list=], return.
+          </li>
+          <li>[=list/For each=] |entry:ordered map| of |json|["file_handlers"]:
+            <ol>
+              <li>Let |file_handler:ordered map| be [=process a file handler
+              item=] with |entry|, |manifest/start_url|, |manifest/scope|, and
+              |manifest_url|.
+              </li>
+              <li>If |file_handler| is failure, continue.
+              </li>
+              <li>[=list/Append=] |file_handler| to |processedFileHandlers|.
+              </li>
+            </ol>
+          </li>
+        </ol>
+        <p>
+          On [=installation=], a user agent SHOULD run the process to
+          [=register file handlers=]
+        </p>
+        <section>
+          <h3>
+            Registering file handlers
+          </h3>
+          <p>
+            The process to <dfn>register file handlers</dfn> on an operating
+            system should be consistent with:
+          </p>
+          <ul>
+            <li>The [=file handler/icon=] and [=file handler/name=] appear in
+            the host operating system surfaces presenting options for opening
+            files, for example within "Open with..." context menus for
+            associated file types.
+            </li>
+            <li>The [=file handler/icon=] appears as the icon of associated
+            files in the host operating system.
+            </li>
+            <li>If there are multiple registered handlers for a file, the OS
+            allows the user to select which app should open it, and also allows
+            the user to set a default app.
+            </li>
+          </ul>
+          <p>
+            A user agent MAY prevent registered file handlers from registering
+            as the 'default' file handler for a given type if the operating
+            system allows.
+          </p>
+        </section>
+        <section>
+          <h3>
+            Privacy consideration: Default [=file handler=].
+          </h3>
+          <p>
+            Depending on the operating system capabilities, the [=manifest/file
+            handler=] could become a 'default' handler (e.g. handling launches
+            of a given file type automatically) of a given [=file type=]
+            without the explicit knowledge of the user. To protect against
+            possible mis-use, user agents MAY require explicit user consent to
```

Comment:
In general, triggering this API requires user action at the OS level, e.g. `file browser > file > context menu > open with > app`. There's a similar amount of user interaction as with `<input type="file">`. If a user agent is careful about how it presents the app in OS level surfaces and depending on details of the OS (such as whether it might silently make a new app the default handler for a file type) then this added confirmation step is arguably not necessary.

That said, the current plan for Chromium **is** to show this extra user consent step.